specs/protocol/isthmus/exec-engine.md

```diff
@@ -63,6 +80,36 @@ places a burden on users of the system in a post-fault-proofs world, where:
 Placing the [`L2ToL1MessagePasser`][l2-to-l1-mp] account storage root in the `withdrawalsRoot` field alleviates this burden
 for users and protocol participants alike, allowing them to propose and verify other proposals with lower operating costs.
 
+#### Genesis Block
+
+If isthmus is active at the genesis block, the withdrawals root is the empty withdrawals root, regardless of L2 state.
+
+#### State Processing
+
+At the time of state processing, the header for which transactions are being validated should not make it's `withdrawalsRoot`
+available to the EVM/application layer.
+
+#### P2P
+
+During sync, the block body withdrawal hash is computed from the withdrawals list in the block body.
+When verifying the final block that is synced, if Isthmus is active at such a block's header,
+we expect that the computed withdrawal hash must match the hash of an empty list of withdrawals.
+The header `withdrawalsRoot` MPT hash can be any non-null value in this case. Verification of the
+final block is performed after a header is available so that the header's timestamp can be checked
+to see if Isthmus is active.
```

Comment:
I've got to admit, I still don't understand this. There are no withdrawals in any block for OP Stack so wouldn't the computed withdrawal root always be the empty hash? And when the final header is received if Isthmus is active shouldn't it have the message passer account storage root?

I know headers and bodies in devp2p are matched up by comparing roots (ie tx root, receipts root and withdrawals root) so during sync if you request a batch of block bodies, you'd expect that they all have an empty withdrawals list since OP Stack doesn't use withdrawals.  To match them up with the headers you'd have to essentially ignore the withdrawalsRoot (or treat the header as if it had an empty withdrawal root). I think this is trying to explain that.

Reply:
I've updated the description based on the conversation yesterday.